pkg/controllers/dynakube/dynatraceclient/builder.go

```diff
@@ -120,7 +120,6 @@ func (dynatraceClientBuilder builder) verifyTokenScopes(dynatraceClient dtclient
 	var err error
 
 	if dynatraceClientBuilder.dynakube.IsTokenScopeVerificationAllowed(timeprovider.New()) {
-		dynaKubeStatus.DynatraceApi.LastTokenScopeRequest = metav1.Now()
 		err = dynatraceClientBuilder.tokens.VerifyScopes(dynatraceClient)
 		log.Info("token verified")
 	} else {
```

Comment:
the problem here is that timestamp now is always reset if there is no error
- it should only be reset IF `VerifyScopes` was called without any error

this code is a bit confusing because it tries to save 1 `if err != nil` check, but goes about it the wrong way, the whole structure is a bit over complicated

it could be simplified: (super quick github-based code copy-pasta, soo may not work :D)
```
func (dynatraceClientBuilder builder) verifyTokenScopes(dynatraceClient dtclient.Client, dynaKubeStatus *dynatracev1beta1.DynaKubeStatus) error {
	if dynatraceClientBuilder.dynakube.IsTokenScopeVerificationAllowed(timeprovider.New()) {
		dynaKubeStatus.DynatraceApi.LastTokenScopeRequest = metav1.Now()
		err := dynatraceClientBuilder.tokens.VerifyScopes(dynatraceClient)
		if err != nil { 
		     return err
		}
		log.Info("token verified")
		dynaKubeStatus.DynatraceApi.LastTokenScopeRequest = metav1.Now()
		return nil
	}
	log.Info(dynatracev1beta1.GetCacheValidMessage(
			"token verification",
			dynaKubeStatus.DynatraceApi.LastTokenScopeRequest,
			dynatraceClientBuilder.dynakube.FeatureApiRequestThreshold()))
	return lastErrorFromCondition(dynaKubeStatus)
}
```

wdyt?

Reply:
now that I reread it, it may make sense to have the "simple" block of code in the `if statement` and the "complicated" block outside the `if statement` 🤔 

wdyt?